pkg/registry/apps/statefulset/strategy_test.go

```diff
@@ -51,10 +51,11 @@ func TestStatefulSetStrategy(t *testing.T) {
 	ps := &apps.StatefulSet{
 		ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
 		Spec: apps.StatefulSetSpec{
-			PodManagementPolicy: apps.OrderedReadyPodManagement,
-			Selector:            &metav1.LabelSelector{MatchLabels: validSelector},
-			Template:            validPodTemplate.Template,
-			UpdateStrategy:      apps.StatefulSetUpdateStrategy{Type: apps.RollingUpdateStatefulSetStrategyType},
+			PodManagementPolicy:               apps.OrderedReadyPodManagement,
+			PersistentVolumeClaimDeletePolicy: apps.Retain,
```

Comment:
Alpha fields should be dropped in PrepareForCreate, PrepareForUpdate when the feature gate is disabled:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-field-in-existing-api-version (step 3)

Reply:
Done. I had done things through DropDisabledFields but I think your was is right as the behavior is slightly different on Create than Update.